tutorial/cloudshell_tutorial.md

```diff
@@ -2,135 +2,203 @@
-access to your GCP credentials, so you won't need to do anything extra to
+access to your GCP credentials, so you will not need to do anything extra to
 authenticate and start provisioning resources. When using Terraform from another
-environment, you'll need to configure authentication. You can [read about
-credentials
-here](https://www.terraform.io/docs/providers/google/provider_reference.html#credentials).
+environment, you will need to configure authentication. You can read about
+credentials in the [GCP provider
+documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#authentication).
 
-## Terraform Configuration
+## Build Infrastructure
 
 The set of files used to describe infrastructure in Terraform is known as a
-Terraform _configuration_. We're going to write our first configuration now to
-launch a single GCP instance.
+Terraform _configuration_. You will now write your first configuration to create
+a network.
 
-The format of the configuration files is [documented
-here](https://www.terraform.io/docs/configuration/index.html). Configuration
-files can [also be
-JSON](https://www.terraform.io/docs/configuration/syntax.html), but we recommend
-only using JSON when the configuration is generated by a machine.
+Each Terraform configuration must be in its own working directory. Your Cloud
+Shell environment includes a directory called `tutorial` that will store the
+configuration you will use for this tutorial.
 
-The entire configuration is shown below. We'll go over each part soon.
+Terraform loads all files ending in `.tf` or `.tf.json` in the working
+directory.
 
-When run, Terraform will load all configuration files from the current directory. So it's a good idea to have a separate directory for each project. Your Cloud Shell environment should include a directory called `tutorial` that will store the configuration we'll use for this guide.
+Inside of the `tutorial` directory you will find a file named <walkthrough-editor-open-file filePath="terraform-getting-started-gcp-cloud-shell/tutorial/main.tf">main.tf</walkthrough-editor-open-file>.
 
-Inside of it there should be a file named `main.tf`. Terraform recognizes files ending in `.tf` or `.tf.json` as configuration files and will load them when it runs.
+Open `main.tf` in the Cloud Shell editor, and paste in the configuration below.
 
-<walkthrough-editor-open-file filePath="terraform-getting-started-gcp-cloud-shell/tutorial/main.tf">main.tf</walkthrough-editor-open-file>
+```hcl
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "3.73.0"
+    }
+  }
 
-First, we'll configure the provider. Add the following to `main.tf`:
+  required_version = ">= 0.15.0"
+}
 
-```hcl
 provider "google" {
-  version = "3.5.0"
-
   project = "{{project-id}}"
   region  = "us-central1"
   zone    = "us-central1-c"
 }
+
+module "project_services" {
```

Comment:
is there a reason why you chose to use this module? it might be kinda advanced for a getting started guide.

our gcp getting started guide uses:

```
resource "google_compute_network" "vpc_network" {
  name = "terraform-network"
}
```

Reply:
The [GCP getting started tutorial](https://learn.hashicorp.com/tutorials/terraform/google-cloud-platform-build?in=terraform/gcp-get-started#set-up-gcp) has the user set up the services manually in the console before starting the tutorial. This version used to use the `google_project_services` resource so the user didn't have to switch context out of the cloud shell, but that resource has been deprecated. The [upgrade docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_3_upgrade#google_project_services-has-been-removed-from-the-provider) recommends using the module.